model/project_parser.go

```diff
@@ -667,13 +662,26 @@ func evaluateBuildVariants(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluato
 				bvTasks[t.Name] = struct{}{}
 			}
 		}
+
+		// save display task if it contains valid execution tasks
 		for _, dt := range pbv.DisplayTasks {
 			projectDt := DisplayTask{Name: dt.Name}
 			if _, exists := bvTasks[dt.Name]; exists {
 				errs = append(errs, fmt.Errorf("display task %s cannot have the same name as an execution task", dt.Name))
 				continue
 			}
+
+			//resolve tags for display tasks
```

Comment:
Why did this move down?

Reply:
Before we called evalSelector on ParserProject's build execution tasks and then destructively modified the execution tasks, in order to use them in this later loop. However, the execution tasks aren't used in between. By moving the evaluating down into this loop we could simply use the tasks slice rather than modifying parser project.